CRM/Contact/Form/DedupeRules.php

```diff
@@ -60,8 +60,8 @@ public function preProcess() {
     $this->_rgid = CRM_Utils_Request::retrieve('id', 'Positive', $this, FALSE, 0);
     $contactTypes = civicrm_api3('Contact', 'getOptions', array('field' => "contact_type"));
     $contactType = CRM_Utils_Request::retrieve('contact_type', 'String', $this, FALSE, 0);
-    if (in_array($contactType, $contactTypes['values'])) {
-      $this->_contactType = $contactTypes['values'][$contactType];
+    if (CRM_Utils_Array::value($contactType, $contactTypes['values'])) {
```

Comment:
Why the change here? It changes the logic from asking "is $contactType in the array $values" to "is $values[$contactType] truthy?"

Reply:
@mickadoo Michael if you look at how old L64 was crafted it was using it as a Key whereas in_array checks on the value so somewhere we stuffed up. I'm not sure if should be on the key or not

Reply:
so the effect here is that IF contactType is passed in via the URL  we accept it if valid. If not we calculate from the rule perhaps keep this & change lower down

Remove
    elseif (!empty($contactType)) {
      throw new CRM_Core_Exception('Contact Type is Not valid');
    }

& replace with 

    if (!$this->_rgid && !$this->_contactType) {
      CRM_Core_Session::setStatus('Contact Type could not be determined, assuming Individual');
     $this->_contactType = 'Individual';
}

Or we could throw an exception - but I can't see why we don't just use a default